src/i2c/slave.rs

```diff
@@ -350,40 +402,89 @@ impl I2cSlave<'_, Blocking> {
             }
 
             if !stat.slvstate().is_slave_receive() {
-                return Err(TransferError::ReadFail.into());
+                return Err(TransferError::OtherBusError.into());
             }
 
             // Now we can safely read the next byte
             *b = i2c.slvdat().read().data().bits();
-            i2c.slvctl().write(|w| w.slvcontinue().continue_());
             xfer_count += 1;
         }
 
         let stat = i2c.stat().read();
         if stat.slvdesel().is_deselected() {
-            // Clear the deselect bit
-            i2c.stat().write(|w| w.slvdesel().deselected());
-            return Ok(Response::Complete(xfer_count));
+            // inhibit drop
+            core::mem::forget(self);
```

Comment:
Nit-pick: I would replace with a destructuring assignment.
```suggestion
            let BlockingI2cSlaveWrite{..} = self;
```

Or alternatively have a `Self::defuse` method.